subprojects/junit-jupiter/src/main/java/org/mockito/junit/jupiter/ConfiguredWithMockito.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2018 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito.junit.jupiter;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.quality.Strictness;
+
+import java.lang.annotation.Retention;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Annotation that can configure Mockito as invoked by the {@link MockitoExtension}.
+ */
+@ExtendWith(MockitoExtension.class)
+@Retention(RUNTIME)
+public @interface ConfiguredWithMockito {
```

Comment:
Can you update PR description with all the public API changes / additions so that the team can review them in one spot?

I'm curious why did you go with ```ConfiguredWithMockito```? Alternatives:
 - ```ExtendWithMockito``` - is syntactically consistent with ```ExtendWith```, default Jupiter annotation.
 - ```MockitoExtension.Settings``` - easier to discover, clearer intent (it's a setting for the extension, not general purpose Mockito configuration)
- ```MockitoExtensionSettings```

+1 to ```ExtendWithMockito```
-1 to ```ConfiguredWithMockito```

If I missed a discussion about it, please let me know so that I can read it and revise my vote.

Reply:
This was mostly after a discussion with @marcphilipp We concluded that this annotation is for "configuring" the test, as it takes configuration parameters of the extension. Therefore I would say this is more clear

Reply:
Are you attached to “ConfiguredWithMockito” annotation name? It looks odd in test. Isn’t “WithMockito” or “MockitoSettings” nicer? “Configured” is redundant, “MockitoSettings” is nice because we already have withSettings() method at mock creation. I'm not against "ConfiguredWithMockito" but I think we could do better. I'm looking for a best name because long term we will have similar annotation in mockito-core.